src/runtime_src/core/pcie/common/memaccess.h

```diff
@@ -164,37 +164,39 @@ namespace xcldev {
      * Caller's responsibility to do sanity checks. No sanity checks done here
      */
     int readBank(std::ofstream& aOutFile, unsigned long long aStartAddr, unsigned long long aSize) {
-      char *buf = 0;
+      auto buf = std::make_shared<char>();
       unsigned long long blockSize = 0x20000;
-      if (xrt_core::posix_memalign((void**)&buf, getpagesize(), blockSize))
+      if (xrt_core::posix_memalign((void**)buf.get(), getpagesize(), blockSize))
         return -1;
-      std::memset(buf, 0, blockSize);
+      std::memset(buf.get(), 0, blockSize);
 
       size_t count = aSize;
       uint64_t incr;
+      std::ios_base::fmtflags f(std::cout.flags());
```

Comment:
Nitpicking, you should use an automatic object for this, so that destructor automatically restores the fmt.

Reply:
I have come across this in almost all the files. Right now I am making an RAII class in memaccess.h and in xbmgmt.h (all xbmgmt sub commands are using this now). I have seen this in xbutil as well, but I don't want to make that change in this PR. Is there a better place I can put this class in?